File: .ref-BSD-4_3_Reno/usr/src/lib/libF77/c_cos.c
/*
 * Copyright (c) 1980 Regents of the University of California.
 * All rights reserved.  The Berkeley software License Agreement
 * specifies the terms and conditions for redistribution.
 *
 *	@(#)c_cos.c	5.1	6/7/85
 */

#include "complex"

c_cos(r, z)
complex *r, *z;
{
double sin(), cos(), sinh(), cosh();

r->real = cos(z->real) * cosh(z->imag);
r->imag = - sin(z->real) * sinh(z->imag);
}

File: .ref-Research-V6/usr/source/fort/rt1/ccos.f
	function ccos(z1)
	complex ccos, z1, z2
	dimension dummy(2)
	equivalence (a,z2,dummy(1)), (b,dummy(2))

	z2 = z1
	c = exp(b)
	d = 1.0/c
	b = sin(a)*(d-c)/2.0
	a = cos(a)*(c+d)/2.0
	ccos = z2
	return
	end
